src/tasks/block.rs

```diff
@@ -161,7 +159,11 @@ impl BlockBuilder {
     }
 
     /// Fetches bundles from the cache and ingests them into the in progress block
-    async fn get_bundles(&mut self, host_provider: &Provider, in_progress: &mut InProgressBlock) {
+    async fn get_bundles(
+        &mut self,
+        ru_provider: &WalletlessProvider,
+        in_progress: &mut InProgressBlock,
+    ) {
         // Authenticate before fetching to ensure access to a valid token
         if let Err(err) = self.bundle_poller.authenticator.authenticate().await {
```

Comment:
the authenticator spawns a task which continuously refreshes the token. There is no need to programmatically do so.